File: SIC_Backend/Repositories/ISectionsRepository.cs
using SIC_Backend.Data.DTOs;
using SIC_Backend.Data.Models;

namespace SIC_Backend.Repositories
{
    public interface ISectionsRepository
    {
        public SectionDTO GetSectionById(int id);
        public IEnumerable<SectionDTO> GetSectionsByUserId(string userId);
        public IEnumerable<SectionDTO> GetSectionsByPlaceId(int placeId);
        public Task CreateSectionAsync(CreateSectionModel model);
        public Task UpdateSectionAsync(int id, UpdateSectionModel model);
        public Task DeleteSectionAsync(int id);
    }
}

File: sic_home/lib/repositories/interfaces/i_sections_repository.dart
import 'package:sic_home/models/create_section_model.dart';
import 'package:sic_home/models/section.dart';
import 'package:sic_home/models/update_section_model.dart';

abstract class ISectionsRepository {
  Future<Section> getSectionById(int id);

  Future<List<Section>> getSectionsByUser(String userId);

  Future<List<Section>> getSectionsByPlace(int placeId);

  Future<void> createSection(CreateSectionModel section);

  Future<void> updateSection(int id, UpdateSectionModel section);

  Future<void> deleteSection(int id);
}
